pyccel/naming/ccudanameclashchecker.py

```diff
@@ -0,0 +1,69 @@
+# coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+Handles name clash problems in Ccuda
+"""
+from pyccel.utilities.metaclasses import Singleton
+from pyccel.utilities.strings import create_incremented_string
+
+class CCudaNameClashChecker(metaclass = Singleton):
+    """ Class containing functions to help avoid problematic names in Ccuda
+    """
+    # Keywords as mentioned on https://en.cppreference.com/w/c/keyword
```

Comment:
we still need to add some of these : https://icl.utk.edu/~mgates3/docs/cuda.html